src/go/k8s/apis/redpanda/v1alpha1/cluster_types.go

```diff
@@ -798,6 +799,20 @@ func (r *Cluster) PandaproxyAPITLS() *PandaproxyAPI {
 	return nil
 }
 
+// SchemaRegistryAPIURL returns a SchemaRegistry URL string.
+// It returns internal URL unless externally available with TLS.
+func (r *Cluster) SchemaRegistryAPIURL() string {
+	host := r.Status.Nodes.SchemaRegistry.Internal
+	if r.IsSchemaRegistryExternallyAvailable() && r.IsSchemaRegistryTLSEnabled() {
+		host = r.Status.Nodes.SchemaRegistry.External
```

Comment:
what if the cluster is not started yet and the status in urls is empty?

Reply:
:+1: I updated to requeue if `Cluster` status is not yet configured [REF](https://github.com/redpanda-data/redpanda/pull/5758/commits/ace25cd1e24aa2898eb692cfb905dc423ca56068)

Reply:
Marking this as resolved [REF](https://github.com/redpanda-data/redpanda/pull/5758/commits/5b0c9a9d02253067a7c31108ac379de8ac30375e) The ref above might have been force pushed